KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  NewCheckoutPresenter.swift
+//  KarhooUISDK
+//
+//  Created by Aleksander Wedrychowski on 05/01/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+import UIKit
+import SwiftUI
+
+enum NewCheckoutState {
```

Comment:
This looks new. Why was it needed? How does the ride being scheduled or asap affect the UI or the flow apart from a some text in a label? Also when is `loading` used? When the booking request is being sent to the backend or when we're waiting for the loyalty status to be refreshed? Some other scenario?

Reply:
This is an optional idea, shall be implemented if during the actual implementation we agree the state introduction is needed. For now, it's not used at all - I added it only to make us aware about the potential approach.